File: herald-admin-typescript/src/namespaces/blocks.ts
import type { HttpTransport } from "../transport.js";

export class BlockNamespace {
  constructor(private transport: HttpTransport) {}

  async block(userId: string, blockedId: string): Promise<void> {
    await this.transport.request("POST", "/blocks", { user_id: userId, blocked_id: blockedId });
  }

  async unblock(userId: string, blockedId: string): Promise<void> {
    await this.transport.request("DELETE", "/blocks", { user_id: userId, blocked_id: blockedId });
  }

  async list(userId: string): Promise<string[]> {
    const res = await this.transport.request<{ blocked: string[] }>("GET", `/blocks/${encodeURIComponent(userId)}`);
    return res.blocked;
  }
}

File: herald-admin-ruby/lib/herald_admin/blocks.rb
# frozen_string_literal: true

module HeraldAdmin
  class BlockNamespace
    def initialize(transport)
      @t = transport
    end

    def block(user_id, blocked_id)
      @t.request("POST", "/blocks", { user_id: user_id, blocked_id: blocked_id })
    end

    def unblock(user_id, blocked_id)
      @t.request("DELETE", "/blocks", { user_id: user_id, blocked_id: blocked_id })
    end

    def list(user_id)
      data = @t.request("GET", "/blocks/#{ERB::Util.url_encode(user_id)}")
      data["blocked"]
    end
  end
end
